packages/tools/index.ts

```diff
@@ -0,0 +1,52 @@
+import asc from "assemblyscript/bin/asc";
+import fs from "fs";
+import { BaseTool } from "./assembly/base";
+
+type ToolInfo = {
+  id: string;
+  name: string;
+};
+
+export interface IToolFactory {
+  getToolList(): ToolInfo[];
+  downloadTool(toolId: string, storageDir?: string): void;
+  loadTool(filePath: string): Promise<BaseTool>;
+}
+
+export default class ToolFactory implements IToolFactory {
+  getToolList(): ToolInfo[] {
+    return [
+      {
+        id: "query-engine-tool",
+        name: "Query Engine Tool",
+      },
+      {
+        id: "wiki-tool",
+        name: "Wikipedia Summary",
+      },
+    ];
+  }
+
+  async downloadTool(toolId: string, storageDir: string = "build") {
+    const { error, stdout, stderr } = await asc.main([
```

Comment:
isn't that starting the compiler?
the build process should compile the `wasm` files for each tool and then the file should be just loaded here